adal/src/main/java/com/microsoft/aad/adal/Utility.java

```diff
@@ -46,4 +50,10 @@ static boolean isClaimsChallengePresent(final AuthenticationRequest request) {
         // if developer pass claims down through extra qp, we should also skip cache.
         return !StringExtensions.isNullOrBlank(request.getClaimsChallenge());
     }
+
+    static URL constructAuthorityUrl(final URL originalAuthority, final String host) throws MalformedURLException {
+        final String path = originalAuthority.getPath().replaceFirst("/", "");
+        final Uri.Builder builder = new Uri.Builder().scheme(originalAuthority.getProtocol()).authority(host).appendPath(path);
+        return new URL(builder.build().toString());
```

Comment:
nit: you can just do builder.build().toURL

Reply:
another nit: there was some thing about commenting static routines.

Reply:
can you elaborate a bit more on commenting static routines?

Reply:
I don't think i can do a toURL with Uri.builder.build